crates/solana-gateway-task-processor/src/component.rs

```diff
@@ -147,511 +144,225 @@ async fn process_task(
-                let ix = instructions::initialize_pending_command(
-                    &self.metadata.gateway_root_pda,
-                    &self.signer,
-                    command,
-                )
-                .ok()?;
-
-                let output = async move {
-                    let send_transaction_result =
-                        send_transaction(self.solana_rpc_client, self.keypair, ix).await;
-
-                    let Err(err) = send_transaction_result else {
-                        // tx was successfully executed
-                        return Ok(approved_message_pda)
-                    };
-
-                    // tx was not executed -- inspect root cause
-                    let ComputeBudgetError::SimulationError(ref simulation) = err else {
-                        // some kid of irrecoverable error
-                        return Err(eyre::Error::from(err))
-                    };
-
-                    // this is the error for when a message PDA was already registered
-                    if matches!(
-                        simulation.err,
-                        Some(TransactionError::InstructionError(
-                            1, // <-- 0th idx is the ComputeBudget prefix
-                            InstructionError::InvalidAccountData
-                        ))
-                    ) {
-                        return eyre::Result::Ok(approved_message_pda);
-                    }
-
-                    // Return the simulation error
-                    Err(eyre::Error::from(err))
+    solana_rpc_client: &RpcClient,
+    keypair: &Keypair,
+) -> Result<(), eyre::Error> {
+    let payload = execute_task.payload;
+
+    // compose the message
+    let message = Message {
+        cc_id: CrossChainId {
+            chain: execute_task.message.source_chain,
+            id: execute_task.message.message_id.0,
+        },
+        source_address: execute_task.message.source_address,
+        destination_chain: metadata.name_of_the_solana_chain.clone(),
+        destination_address: execute_task.message.destination_address,
+        payload_hash: execute_task
+            .message
+            .payload_hash
+            .try_into()
+            .unwrap_or_default(),
+    };
+    let command_id = command_id(&message.cc_id.chain, &message.cc_id.id);
+    let (gateway_incoming_message_pda, ..) =
+        axelar_solana_gateway::get_incoming_message_pda(&command_id);
+
+    // For compatibility reasons with the rest of the Axelar protocol we need add custom handling
+    // for ITS & Governance programs
+    let destination_address = message.destination_address.parse::<Pubkey>()?;
+    match destination_address {
+        axelar_solana_its::ID => {
+            // todo ITS specific handling
+            tracing::error!("ITS program not yet supported");
+        }
+        axelar_solana_governance::ID => {
+            // todo Governance specific handling
```

Comment:
@eloylp this is for this issue -- https://github.com/eigerco/axelar-relayer-core-internal/issues/28

Reply:
ack, thank you